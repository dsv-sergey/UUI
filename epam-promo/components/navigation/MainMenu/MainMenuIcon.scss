@use  '../../../assets/styles/index' as *;
@import '../../../assets/styles/colorvars/navigation/mainMenu/mainMenuIcon-colorvars.scss';

.container {
    @include mainMenu-icon-color();
    width: 60px;
    height: 60px;
    display: flex;
    justify-content: center;
    flex-shrink: 0;
    cursor: pointer;

    &:hover {
        background: var(--background-color-hover);
    }

    &:active {
        background-color: var(--background-color-active);
    }
}
