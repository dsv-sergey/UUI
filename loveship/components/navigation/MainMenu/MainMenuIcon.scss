@use '../../../assets/styles/scss/index'  as *;

.container {
    width: 60px;
    height: 60px;
    display: flex;
    justify-content: center;
    flex-shrink: 0;
    cursor: pointer;

    &:hover {
        background: $night900;
    }

    &:active {
        background-color: darken($night900, 10%);
    }
}
