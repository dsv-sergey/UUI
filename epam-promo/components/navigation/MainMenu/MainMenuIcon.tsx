import * as React from 'react';
import { Icon, IClickable, IAdaptiveItem } from '@epam/uui';
import { ButtonProps, IconContainer } from '@epam/uui-components'; 
import * as css from './MainMenuIcon.scss';

export interface MainMenuIconProps extends ButtonProps, IAdaptiveItem {
    icon: Icon;
}

export const MainMenuIcon = (props: MainMenuIconProps) => (
    <div onClick={ props.onClick } className={ css.container }>
        <IconContainer icon={ props.icon } />
    </div>
);