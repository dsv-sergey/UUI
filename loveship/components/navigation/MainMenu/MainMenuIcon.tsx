import React from 'react';
import { Icon, IAdaptiveItem } from '@epam/uui';
import { ButtonProps, IconContainer } from '@epam/uui-components';
import cx from 'classnames';
import * as css from './MainMenuIcon.scss';

export interface MainMenuIconProps extends ButtonProps, IAdaptiveItem {
    icon: Icon;
}

export const MainMenuIcon = (props: MainMenuIconProps) => (
    <div onClick={ props.onClick } className={ cx(props.cx, css.container) }>
        <IconContainer icon={ props.icon } />
    </div>
);
