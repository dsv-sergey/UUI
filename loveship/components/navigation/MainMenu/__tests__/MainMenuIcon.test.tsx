import * as React from "react";
import renderer from "react-test-renderer";
import {MainMenuIcon} from "../MainMenuIcon";
import * as acceptIcon from "../../../icons/accept-12.svg";

describe("MainMenuIcon", () => {
    it("should be rendered correctly", () => {
        const tree = renderer
            .create(<MainMenuIcon icon={ acceptIcon }/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<MainMenuIcon
                icon={ acceptIcon }
                target='_blank'
                link={ { pathname: '/' } }
                collapseToMore
                estimatedWidth={ 120 }
                priority={ 6 }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});