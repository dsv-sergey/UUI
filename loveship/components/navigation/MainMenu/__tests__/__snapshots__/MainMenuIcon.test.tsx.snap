// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuIcon should be rendered correctly 1`] = `
<div
  className="container"
>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#accept-12.svg"
      />
    </svg>
  </div>
</div>
`;

exports[`MainMenuIcon should be rendered correctly with props 1`] = `
<div
  className="container"
>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#accept-12.svg"
      />
    </svg>
  </div>
</div>
`;
